### DISCUSSION TITLE
Prevent secret logging

### DISCUSSION THREAD

**Discussion by MaelAbgrall** [axios/axios#5815] [Q&A]
Heyo

I'm using Axios in my backend, and sometimes there are a few errors. Nothing problematic since everything is in try-catch clauses, but I realised that logging the trace for Axios can leak secrets, so I'm only logging the trace when in dev.

I'm curious if there is a way to log this trace while preventing leaking secrets. Python and FastAPI, for example, had a very handy "secret" type that would replace the string with "******" whenever it was logged.

Is there similar functionality in Axios? A way to indicate a certain variable is a secret and should not be logged in the trace?

Note also that I'm unfortunately dealing with a lot of secrets, so a solution like Python above would be much easier than a bespoke catch clause for every API call I'm making

**Comment by jasonsaayman** [comment]
We are considering some of this. We have already added a filter to support you specifying keys you want removed. Please check the latest docs
